promql/bench_test.go

```diff
@@ -310,6 +351,73 @@ func BenchmarkRangeQuery(b *testing.B) {
 	}
 }
 
+func BenchmarkJoinQuery(b *testing.B) {
+	stor := teststorage.New(b)
+	stor.DisableCompactions() // Don't want auto-compaction disrupting timings.
+	defer stor.Close()
+	opts := promql.EngineOpts{
+		Logger:     nil,
+		Reg:        nil,
+		MaxSamples: 50000000,
+		Timeout:    100 * time.Second,
+	}
+	engine := promqltest.NewTestEngineWithOpts(b, opts)
+
+	const interval = 10000 // 10s interval.
+	// A day of data plus 10k steps.
+	numIntervals := 8640 + 10000
+
+	err := setupJoinQueryTestData(stor, engine, interval, numIntervals, 1000)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	cases := []benchCase{
+		{
+			expr:  `rpc_request_success_total + rpc_request_error_total`,
+			steps: 10000,
+		},
+		{
+			expr:  `rpc_request_success_total + ON (job, instance) GROUP_LEFT rpc_request_error_total`,
+			steps: 10000,
+		},
+		{
+			expr:  `rpc_request_success_total AND rpc_request_error_total{instance=~"0.*"}`,
+			steps: 10000,
+		},
+		{
+			expr:  `rpc_request_success_total OR rpc_request_error_total{instance=~"0.*"}`,
+			steps: 10000,
+		},
+		{
+			expr:  `rpc_request_success_total UNLESS rpc_request_error_total{instance=~"0.*"}`,
+			steps: 10000,
+		},
+	}
+
+	for _, c := range cases {
+		name := fmt.Sprintf("expr=%s,steps=%d", c.expr, c.steps)
```

Comment:
```suggestion
		name := fmt.Sprintf("expr=%s/steps=%d", c.expr, c.steps)
```

Rationales: It unlocks https://www.bwplotka.dev/2024/go-microbenchmarks-benchstat/#newly-enabled-flow-comparing-efficiency-across-cases (kudos for using key=val syntax!)